avalanche/benchmarks/scenarios/generic_benchmark_creation.py

```diff
@@ -28,35 +31,284 @@
+            )
+            if not is_int_iterable(as_classification_dataset.targets):
+                return False
+        except Exception:
+            return False
+        
+    return True
+
+
+def _manage_legacy_classification_usage(
+    train_datasets: Sequence[GenericSupportedDataset],
+    test_datasets: Sequence[GenericSupportedDataset],
+    other_streams_datasets: Optional[
+        Mapping[str, Sequence[GenericSupportedDataset]]],
+    dataset_factory: Union[
+        DatasetFactory,
+        Literal['check_if_classification']
+    ],
+    benchmark_factory: Union[Callable[
+        [
+            TStreamsUserDict,
+            bool
+        ], TDatasetScenario
+    ], Literal['check_if_classification']]) -> Tuple[
+        DatasetFactory, 
+        Callable[[
+            TStreamsUserDict,
+            bool
+        ], TDatasetScenario]]:
+
+    check_implicit_classification = \
+        dataset_factory == 'check_if_classification' or \
+        benchmark_factory == 'check_if_classification'
+    
+    is_implicit_classification = False
+    if check_implicit_classification:
+        all_datasets_iterables = [
+            train_datasets,
+            test_datasets,
+        ]
+
+        if other_streams_datasets is not None:
+            all_datasets_iterables.extend(other_streams_datasets.values())
+
+        is_implicit_classification = _detect_legacy_classification_usage(
+            itertools.chain(*all_datasets_iterables)
+        )
+
+    if is_implicit_classification:
+        warnings.warn(
+            '`dataset_benchmark` is being called by passing classification '
+            'datasets. It is recommended to switch to '
+            '`dataset_classification_benchmark` to make sure a '
+            '`ClassificationScenario` is returned',
+            DeprecationWarning
+        )
+    
+    dataset_factory_compat: DatasetFactory
+    if dataset_factory == 'check_if_classification':
+        if is_implicit_classification:
+            dataset_factory_compat = make_classification_dataset
+        else:
+            dataset_factory_compat = make_generic_dataset
+    else:
+        dataset_factory_compat = dataset_factory
+    
+    benchmark_factory_compat: Callable[
+        [
+            TStreamsUserDict,
+            bool
+        ], TDatasetScenario
+    ]
+    if benchmark_factory == 'check_if_classification':       
+        if is_implicit_classification:
+            benchmark_factory_compat = \
+                _make_classification_scenario  # type: ignore
+        else:
+            benchmark_factory_compat = _make_generic_scenario
+    else:
+        benchmark_factory_compat = benchmark_factory
+
+    return dataset_factory_compat, benchmark_factory_compat
 
 
 def create_multi_dataset_generic_benchmark(
-    train_datasets: Sequence[SupportedDataset],
-    test_datasets: Sequence[SupportedDataset],
+    train_datasets: Sequence[GenericSupportedDataset],
+    test_datasets: Sequence[GenericSupportedDataset],
```

Comment:
can we simplify this as require an AvalancheDataset? this way we remove the necessity to pass transform/task labels...